homeassistant/data_entry_flow.py

```diff
@@ -527,8 +527,19 @@ async def _async_handle_step(
         ):
             # The flow's progress task was changed, register a callback on it
             async def call_configure() -> None:
-                with suppress(UnknownFlow):
+                if self._progress.get(flow.flow_id) is None:
+                    return
+
+                try:
                     await self._async_configure(flow.flow_id)
+                except Exception:
+                    _LOGGER.exception("Error processing progress task for %s", flow)
+
+                    # Notify frontend to refresh before aborting
+                    flow.async_notify_flow_changed()
+
+                    with suppress(UnknownFlow):
+                        self.async_abort(flow.flow_id)
 
             def schedule_configure(_: asyncio.Task) -> None:
                 self.hass.async_create_task(call_configure())
```

Comment:
TLDR: if the coroutine raises before yielding, it raises synchronously into the callback context and vanishes.

So what I think what is happening is that anything that raises in a `add_done_callback` always gets silently swallowed.

Since `async_create_task` creates eagerly the exception can raise synchronously which means it raises to the outer context of the task its not considered an unhandled task exception, and since `schedule_configure` is called from `add_done_callback` it just gets silently swallowed.

Reply:
Is it possible we have an exception or failure in `_async_loop_exception_handler` formatting the log which prevents it from showing ?
https://github.com/home-assistant/core/blob/ddd8cf7fde05235ed7f9677c5b7cf858116e50df/homeassistant/runner.py#L211